app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -1,27 +1,136 @@
+    private val imageViewModel by viewModels<GrayscaleViewModel>()
+    private val cameraPermission =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { granted ->
+            if (granted) {
+                Toast.makeText(this, "Permissions granted.", Toast.LENGTH_SHORT).show()
+                Log.d(TAG, "Permission was granted successfully.")
+                startCamera()
+            } else {
+                Toast.makeText(this, "Permissions not granted.", Toast.LENGTH_SHORT).show()
+                Log.d(TAG, "Permission was NOT granted.")
+            }
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        imageViewModel.isCameraPermissionGranted = cameraPermissionGranted()
+        tryStartCamera()
+
         setContent {
-            Theme {
+            FeatureDetectAppTheme {
                 Box(Modifier.fillMaxSize()) {
-                    FloatingActionButton(
-                        onClick = { /*TODO*/ },
-                        modifier = Modifier.align(Alignment.BottomEnd)
-                    ) { }
+                    if (imageViewModel.isCameraPermissionGranted) {
+                        imageViewModel.grayscaleBitmap?.let { bitmap ->
+                            Image(
+                                bitmap = bitmap.asImageBitmap(),
+                                modifier = Modifier.fillMaxSize(),
+                                contentDescription = "Grayscale photo"
+                            )
+                        }
+                    } else {
+                        Text(
+                            text = "Camera permission required",
+                            modifier = Modifier.align(Alignment.Center)
+                        )
+                    }
                 }
             }
         }
     }
+
+    override fun onResume() {
+        super.onResume()
+        imageViewModel.isCameraPermissionGranted = cameraPermissionGranted()
+        tryStartCamera()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        cameraExecutor.shutdown()
+    }
+
+    private fun tryStartCamera() = when {
+        imageViewModel.isCameraPermissionGranted -> startCamera()
+        shouldRationalize() -> {
+            Toast.makeText(
+                this,
+                "Without camera permission app can't get and display keypoints.",
+                Toast.LENGTH_SHORT
+            ).show()
+        }
+        else -> cameraPermission.launch(Manifest.permission.CAMERA)
+    }
+
+    private fun cameraPermissionGranted() = ContextCompat.checkSelfPermission(
+        this,
+        Manifest.permission.CAMERA
+    ) == PackageManager.PERMISSION_GRANTED
+
+    private fun shouldRationalize() =
+        (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) &&
+            shouldShowRequestPermissionRationale(Manifest.permission.CAMERA)
```

Comment:
If I press `Crtl+Alt+L` Android Studio adds additional indents here

Reply:
Hm. Ktlint (with `adroid` flag) gives me an error for any other indents.